modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -558,7 +558,7 @@ private boolean isPathParamDefined(String pathParam, List<Parameter> parameters)
             return false;
         } else {
             Parameter pathParamDefined = parameters.stream()
-                            .filter(parameter -> pathParam.equals(parameter.getName()) && "path".equals(parameter.getIn()))
+                            .filter(parameter -> (parameter.get$ref() != null) || (pathParam.equals(parameter.getName()) && "path".equals(parameter.getIn())))
```

Comment:
This checks only having $ref, but does not check that the reference is for this name and contains `in: path`

Reply:
Actually, yep, I agree to fix false matching for valid file even with workaround.

From my pov this kind of validation should be called outside of deserialization, even after resolving all $refs, and only in case when ParseOptions have $refs resolving enabled.
Somewhere here, probably: https://github.com/swagger-api/swagger-parser/blob/77f69e9407890c472f5b7c0b5c2c34d484dac4e5/modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/OpenAPIV3Parser.java#L65

What do you think?

Reply:
Agree.
The root problem is that currently validation and deserialization is tightly coupled and ref-resolve is not considered.
We should de-couple them. validation should be done **after** deserialization.

I quickly read OpenAPIDeserializer and found that many validation is done in it 😢 (easily find with test searching `result.warning(`). 
It seems to cost much work to separate deserialization and validation...
So I think we should create an issue about the decoupling, work on it in the future.

Let me know what you think. :)

Reply:
https://github.com/swagger-api/swagger-parser/issues/982
issued!